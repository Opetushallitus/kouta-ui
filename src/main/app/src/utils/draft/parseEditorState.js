import { convertFromHTML } from 'draft-convert';
import { EditorState } from 'draft-js';
import { isString } from 'lodash';
import linkDecorator from './linkDecorator';
import createEmptyEditorState from './createEmptyEditorState';

const createEditorStateWithContent = contentState =>
  EditorState.createWithContent(contentState, linkDecorator);

const parseEditorstate = value => {
  if (!isString(value)) {
    return createEmptyEditorState();
  }

  return createEditorStateWithContent(convertFromHTML(value));
};

export default parseEditorstate;
