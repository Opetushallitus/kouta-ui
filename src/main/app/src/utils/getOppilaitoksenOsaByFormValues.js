import get from 'lodash/get';
import mapValues from 'lodash/mapValues';

import serializeEditorState from './draft/serializeEditorState';
import { isNumeric } from './index';

const getOppilaitoksenOsaByFormValues = ({ tila, muokkaaja, ...values }) => {
  const { perustiedot, esittely, yhteystiedot } = values;

  const { osoite, postinumero, puhelinnumero, verkkosivu } = yhteystiedot;

  return {
    tila,
    muokkaaja,
    metadata: {
      osoite: osoite || {},
      postinumeroKoodiUri: get(postinumero, 'value') || null,
      puhelinnumero: puhelinnumero || null,
      verkkosivu: verkkosivu || null,
      esittely: mapValues(esittely || {}, serializeEditorState),
      opiskelijoita: isNumeric(get(perustiedot, 'opiskelijoita'))
        ? parseInt(perustiedot.opiskelijoita)
        : null,
      kampus: get(perustiedot, 'kampus') || {},
    },
  };
};

export default getOppilaitoksenOsaByFormValues;
