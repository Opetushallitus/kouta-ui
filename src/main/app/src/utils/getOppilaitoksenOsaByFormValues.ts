import { get, mapValues, pick } from 'lodash';

import serializeEditorState from './draft/serializeEditorState';
import { isNumeric } from './index';

const getOppilaitoksenOsaByFormValues = ({ tila, muokkaaja, ...values }) => {
  const { perustiedot, esittely, yhteystiedot, kieliversiot } = values;

  const {
    osoite,
    postinumero,
    puhelinnumero,
    verkkosivu,
    sahkoposti,
  } = yhteystiedot;

  return {
    tila,
    muokkaaja,
    kielivalinta: kieliversiot,
    metadata: {
      yhteystiedot: {
        osoite: {
          osoite: pick(osoite || {}, kieliversiot),
          postinumeroKoodiUri: get(postinumero, 'value') || null,
        },
        sahkoposti: pick(sahkoposti || {}, kieliversiot),
        puhelinnumero: pick(puhelinnumero || {}, kieliversiot),
        wwwSivu: pick(verkkosivu || {}, kieliversiot),
      },
      esittely: mapValues(
        pick(esittely || {}, kieliversiot),
        serializeEditorState
      ),
      opiskelijoita: isNumeric(get(perustiedot, 'opiskelijoita'))
        ? parseInt(perustiedot.opiskelijoita)
        : null,
      kampus: pick(get(perustiedot, 'kampus') || {}, kieliversiot),
    },
  };
};

export default getOppilaitoksenOsaByFormValues;
