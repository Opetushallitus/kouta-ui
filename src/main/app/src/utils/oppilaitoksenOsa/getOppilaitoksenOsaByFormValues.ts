import _ from 'lodash';

import { isNumeric, toKielistettyWithValueStr } from '#/src/utils';

import {
  pickAndSerializeTranslations,
  getKieleistyksetForKieliversiot,
} from '../pickTranslations';

export const getOppilaitoksenOsaByFormValues = ({
  tila,
  muokkaaja,
  ...values
}) => {
  const {
    perustiedot,
    esittely,
    kieliversiot,
    teemakuvaOrEsittelyvideo,
    esikatselu = false,
    hakijapalveluidenYhteystiedot: hy,
  } = values;
  const kieleistykset = getKieleistyksetForKieliversiot(kieliversiot);
  const hpy = Object.values(kieleistykset(hy?.nimi)).some(
    n => String(n).trim().length > 0
  );
  const composeEsittelyvideoNimiObject = (
    videoUrls: object
  ): object | undefined => {
    const languages = Object.keys(videoUrls).filter(lang =>
      Boolean(videoUrls[lang])
    );
    return languages.length > 0
      ? languages.reduce((obj, lang) => {
          return {
            ...obj,
            [lang]: 'esittelyvideo',
          };
        }, {})
      : undefined;
  };
  return {
    tila,
    muokkaaja,
    kielivalinta: kieliversiot,
    teemakuva:
      teemakuvaOrEsittelyvideo?.mediaType === 'teemakuva' &&
      teemakuvaOrEsittelyvideo?.teemakuvaUrl
        ? teemakuvaOrEsittelyvideo?.teemakuvaUrl
        : undefined,
    esikatselu,
    metadata: {
      esittely: pickAndSerializeTranslations(esittely, kieliversiot),
      opiskelijoita: isNumeric(perustiedot?.opiskelijoita)
        ? parseInt(perustiedot.opiskelijoita)
        : null,
      kampus: kieleistykset(perustiedot?.kampus),
      wwwSivu: _.isEmpty(perustiedot?.wwwSivuUrl)
        ? null
        : {
            url: kieleistykset(perustiedot.wwwSivuUrl),
            nimi: kieleistykset(perustiedot.wwwSivuNimi),
          },
      jarjestaaUrheilijanAmmKoulutusta:
        perustiedot?.jarjestaaUrheilijanAmmKoulutusta,
      hakijapalveluidenYhteystiedot: hpy
        ? {
            nimi: kieleistykset(hy.nimi),
            postiosoite:
              !_.isEmpty(hy.postiosoite) || hy.postinumero
                ? {
                    osoite: kieleistykset(hy.postiosoite),
                    postinumeroKoodiUri: kieleistykset(
                      toKielistettyWithValueStr(hy.postinumero)
                    ),
                  }
                : null,
            kayntiosoite:
              !_.isEmpty(hy.kayntiosoite) || hy.kayntiosoitePostinumero
                ? {
                    osoite: kieleistykset(hy.kayntiosoite),
                    postinumeroKoodiUri: kieleistykset(
                      toKielistettyWithValueStr(hy.kayntiosoitePostinumero)
                    ),
                  }
                : null,
            sahkoposti: kieleistykset(hy.sahkoposti),
            puhelinnumero: kieleistykset(hy.puhelinnumero),
          }
        : null,
      esittelyvideo:
        teemakuvaOrEsittelyvideo?.mediaType === 'esittelyvideo' &&
        Object.values(teemakuvaOrEsittelyvideo?.esittelyvideoUrl || {}).filter(
          Boolean
        ).length > 0
          ? {
              url: kieleistykset(teemakuvaOrEsittelyvideo?.esittelyvideoUrl),
              nimi: composeEsittelyvideoNimiObject(
                teemakuvaOrEsittelyvideo?.esittelyvideoUrl || {}
              ),
            }
          : undefined,
    },
  };
};
