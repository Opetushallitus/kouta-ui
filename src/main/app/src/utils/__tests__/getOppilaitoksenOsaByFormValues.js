import getOppilaitoksenOsaByFormValues from '../getOppilaitoksenOsaByFormValues';
import parseEditorState from '../draft/parseEditorState';

test('getOppilaitoksenOsaByFormValues returns correct oppilaitoksen osa given form values', () => {
  const oppilaitos = getOppilaitoksenOsaByFormValues({
    tila: 'tallennettu',
    muokkaaja: '1.1.1.1',
    kieliversiot: ['fi', 'sv'],
    esittely: {
      fi: parseEditorState('<i>Fi esittely</i>'),
      sv: parseEditorState('<i>Sv esittely</i>'),
    },
    perustiedot: {
      opiskelijoita: '100',
      kampus: {
        fi: 'Fi kampus',
        sv: 'Sv kampus',
      },
    },
    yhteystiedot: {
      osoite: {
        fi: 'Fi osoite',
        sv: 'Sv osoite',
      },
      postinumero: { value: 'postinumero_1#1' },
      puhelinnumero: '123456',
      verkkosivu: {
        fi: 'www.verkkosivu.fi',
        sv: 'www.verkkosivu.sv',
      },
    },
  });

  expect(oppilaitos).toMatchSnapshot();
});
