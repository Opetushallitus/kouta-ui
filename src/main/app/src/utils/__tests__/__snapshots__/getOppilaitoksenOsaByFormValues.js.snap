// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOppilaitoksenOsaByFormValues returns correct oppilaitoksen osa given form values 1`] = `
Object {
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "metadata": Object {
    "esittely": Object {
      "fi": "<p><em>Fi esittely</em></p>",
      "sv": "<p><em>Sv esittely</em></p>",
    },
    "kampus": Object {
      "fi": "Fi kampus",
      "sv": "Sv kampus",
    },
    "opiskelijoita": 100,
    "osoite": Object {
      "osoite": Object {
        "fi": "Fi osoite",
        "sv": "Sv osoite",
      },
      "postinumeroKoodiUri": "postinumero_1#1",
    },
    "puhelinnumero": "123456",
    "wwwSivu": Object {
      "fi": "www.verkkosivu.fi",
      "sv": "www.verkkosivu.sv",
    },
  },
  "muokkaaja": "1.1.1.1",
  "tila": "tallennettu",
}
`;
